### ISSUE TITLE
test(ggufload): bind Strix Halo performance work to the pinned Qwen3.8 27B Q4_K_M artifact

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13314]
# test(ggufload): bind Strix Halo performance work to the pinned Qwen3.8 27B Q4_K_M artifact

```routing
repo: fak
lane: ggufload
paths: ["internal/ggufload/qwen38_q4km_artifact_test.go", "internal/ggufload/testdata/qwen38_q4km_header.gguf.gz"]
expected_steps: 3
github_issue: anthony-chaudhary/fak#13314
```

## Working spine

Pinned artifact header -> production GGUF parser -> loaded `model.Config` and tensor inventory -> explicit native quality/admission verdict.

## Current state

Generic Qwen3.8 architecture aliases and synthetic next-token fixtures already exist. They do not prove that the production loader accepts the actual `Qwen3.8-27B-Q4_K_M.gguf` artifact (17,106,775,008 bytes, SHA-256 `7e78da5d7e3ae28d178121f58646953305f3e5bd3cb46f4a75584e8b6c6fe169`) with its real metadata and tensor names. Prior receipts establish this artifact identity; this leaf converts its scrubbed header into a deterministic loader contract. No new alias is indicated.

## Why this is next

Every Halo performance claim depends on loading the intended artifact through fak-native code. A pinned artifact contract prevents later memory and Vulkan work from optimizing a synthetic or misclassified configuration.

## Parent context

Parent: anthony-chaudhary/fak#11572 (existing Halo native campaign).

Qwen3.8 Halo native plan; parent contribution 3/40. Existing generic alias coverage remains owned by the current Qwen GGUF tests.

## Core through-line

Add a scrubbed header fixture derived from the pinned Q4_K_M artifact and a focused test that reads it through the production parser, checks extraction provenance and fixture identity, checks the loaded hybrid geometry, inspects whether the GGUF actually contains its declared MTP inventory, and refuses a mutated or incomplete quality-critical contract.

## Gold-plating boundary

No new Qwen alias, loader rewrite, weight payload copy, network fetch, benchmark, kernel change, or claimed Qwen3.8 compatibility beyond this exact artifact. Do not duplicate generic tests in `gguf_qwen38_arch_test.go`.

## Exact seams and reuse decision

- Reuse `internal/ggufload/gguf_config.go` and the existing Qwen3.8/Qwen3.5 config path; do not create a parallel parser.
- Reuse the pinned identity already recorded under `docs/_witnesses/issue-9525-qwen38-sequence-prefill/`.
- The new fixture contains header metadata only, no model weights or private path. Its own digest is distinct from the source artifact digest; the extraction record names the independently verified source SHA-256 rather than pretending the header can reproduce it.

## Likely files

- `internal/ggufload/qwen38_q4km_artifact_test.go` (test intended to add)
- `internal/ggufload/testdata/qwen38_q4km_header.gguf.gz` (scrubbed fixture intended to add)

## Three bounded steps

1. Derive and checksum the scrubbed header fixture from the independently verified pinned artifact, recording both distinct digests.
2. Add one table-driven production-parser contract covering identity, config, tensor inventory, and a corrupted quality guard.
3. Run the focused package witness and record its exact result.

## Done condition / witness

From the public `fak` root:

## Witness

```powershell
go test -count=1 ./internal/ggufload -run 'TestQwen38Q4KMArtifactContract'
```
Witness: `go test -count=1 ./internal/ggufload -run 'TestQwen38Q4KMArtifactContract'`

## Definition of done

- [ ] Repro witness loads the scrubbed header through production GGUF parsing, not a hand-built `model.Config`.
- [ ] File:line seam stays within the two declared `internal/ggufload` paths.
- [ ] Blast radius is limited to GGUF Qwen admission tests; runtime behavior is unchanged.
- [ ] Quarantined fallback is explicit: absent/mismatched artifact evidence remains `NOT_READY`, never inferred from aliases.
- [ ] Scoped acceptance checks extraction provenance, fixture SHA-256, 64 target layers, 3-linear:1-full cadence, vocabulary 248320, gate metadata, required tensor families, and the observed MTP inventory; a declared-but-absent MTP payload is typed unsupported rather than invented.

## Acceptance gate

The focused test passes from a clean public checkout, the fixture contains no weights or private coordinates, and mutation cases fail closed. Passing this gate proves only exact-artifact loader admission, not logits parity or performance.

## Closure binding

Resolving commit cites the eventual public issue and carries `(fak ggufload)`.

## Lane

`ggufload`

## Expected steps

3

## Value

- Centrality: Core
- P1 Context: advanced - binds later work to the real artifact
- P2 Net value: preserved - no performance claim
- P3 Adaptation: preserved - existing loader and aliases remain unchanged
- P4 Operations: advanced - artifact mismatch becomes an explicit refusal

### Model tier

- Required model tier: `tier/T1-required`
- Optimal model tier: `tier/T1-optimal`

## Work estimate

Estimate: 3 points

## Overall completion contribution

Contribution: 3/40 points

## Completion standard

development

github_issue: anthony-chaudhary/fak#13314

## Tracking

Issue: https://github.com/anthony-chaudhary/fak/issues/13314

**Comment by anthony-chaudhary** [comment]
Resolved in commit 81ec2add9479807603b787dad2131d9885d34a0f (pushed a8b09f27dc..81ec2add94 main -> main).

Verification witness: \go test -count=1 ./internal/ggufload -run 'TestQwen38Q4KMArtifactContract' -v\ -> PASS (contract + 3 fail-closed subtests); \go test -count=1 ./internal/ggufload/...\ -> ok (12.5s); \go vet ./internal/ggufload/\ -> clean; \gofmt -l\ -> clean; 5-gate boundary (targeted public file) -> PASS 0 violations; public leak audit -> clean.

Fixture: losslessly compressed header prefix (10,996,704 bytes decompressed; header SHA-256 2b388fac...) through the 32-byte-aligned tensor-data offset, no tensor payloads. Source artifact identity recorded in-test: 17,106,775,008 bytes, SHA-256 7e78da5d7e3ae28d178121f58646953305f3e5bd3cb46f4a75584e8b6c6fe169.

Finding: the pinned artifact declares qwen35.nextn_predict_layers=1 and actually ships the trailing blk.64 nextn block, so the declared MTP inventory is present (not absent as the issue hypothesized). [SW-VERIFIED] software-only.
